internal/state/execution.go

```diff
@@ -466,12 +466,12 @@ func buildExtendedCommitInfo(ec *types.ExtendedCommit, store Store, initialHeigh
 		}
 
 		var ext []byte
-		if ecs.BlockIDFlag == types.BlockIDFlagCommit {
-			// We only care about vote extensions if a validator has voted to
-			// commit.
-			ext = ecs.Extension
+		if err := ecs.EnsureExtension(); err != nil && vp.RequireExtensions(ec.Height) {
```

Comment:
Here as well, we should check if we are getting extended votes but we are _below_ the "required extension height"